src/core/execute.c

```diff
@@ -453,6 +453,12 @@ int exec_spawn(
          * handoff timestamp. */
         dual_timestamp_now(&start_timestamp);
 
+        r = 0;
+        if (unit->manager->starting_ambient_set != 0 && getpid_cached() != 1)
```

Comment:
This seems overly complicated, all we need is to keep the current behaviour for generators, which can achieved by simply dropping caps after the fork in `execute_strv()`. A new flag for `safe_fork()` passed from the right caller would be enough.

Reply:
My approach (after we've established that we don't want to pass the ambient set to generators) is that it shouldn't be passed to anything but sd-executor. Should a new functionality involving execution of external binaries be introduced, with your approach (as I understand it) its author would need to be of the ambient set and drop it. With mine, however, they need to "restore" the set if it's needed and don't bother at all if not.

Reply:
Yeah but we don't do that often, and a flag in safe_fork is enough to ensure it is used where needed. Moving this stuff around so much is too complex.